src/main/java/org/elasticsearch/plugin/readonlyrest/acl/blocks/rules/impl/SearchlogSyncRule.java

```diff
@@ -17,48 +17,22 @@
 
 package org.elasticsearch.plugin.readonlyrest.acl.blocks.rules.impl;
 
-import org.apache.logging.log4j.Logger;
-import org.elasticsearch.action.ActionRequest;
-import org.elasticsearch.action.ActionResponse;
-import org.elasticsearch.action.search.SearchRequest;
-import org.elasticsearch.action.search.SearchResponse;
-import org.elasticsearch.common.bytes.BytesReference;
-import org.elasticsearch.common.logging.Loggers;
-import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.common.xcontent.XContentHelper;
-import org.elasticsearch.plugin.readonlyrest.acl.blocks.BlockExitResult;
 import org.elasticsearch.plugin.readonlyrest.acl.blocks.rules.RuleExitResult;
-import org.elasticsearch.plugin.readonlyrest.acl.blocks.rules.RuleNotConfiguredException;
 import org.elasticsearch.plugin.readonlyrest.acl.blocks.rules.SyncRule;
-import org.elasticsearch.plugin.readonlyrest.wiring.requestcontext.RequestContext;
-
-import java.io.IOException;
-import java.util.Arrays;
-import java.util.Optional;
+import org.elasticsearch.plugin.readonlyrest.requestcontext.RequestContext;
+import org.elasticsearch.plugin.readonlyrest.settings.rules.SearchlogRuleSettings;
 
 /**
  * Created by sscarduzio on 27/03/2017.
  */
 public class SearchlogSyncRule extends SyncRule {
-  private final Logger logger = Loggers.getLogger(getClass());
-
-  private boolean shouldLog = false;
 
-  public SearchlogSyncRule(Settings s) throws RuleNotConfiguredException {
-    try {
-      shouldLog = s.getAsBoolean(getKey(), false);
-      if (!shouldLog) throw new RuleNotConfiguredException();
-    } catch (Exception e) {
-      throw new RuleNotConfiguredException();
-    }
-  }
+  private final boolean shouldLog;
+  private final SearchlogRuleSettings settings;
 
-  public static Optional<SearchlogSyncRule> fromSettings(Settings s) {
-    try {
-      return Optional.of(new SearchlogSyncRule(s));
-    } catch (RuleNotConfiguredException ignored) {
-      return Optional.empty();
-    }
+  public SearchlogSyncRule(SearchlogRuleSettings s) {
+    shouldLog = s.isEnabled();
```

Comment:
Yeah, but  you see, I kinda still miss the one-rule = one-class simplicity.

But now I think I understood why you split the old rule in the "RuleActionListener" + "RuleSettings" + "Rule" triplet.  Tell me if it's true! 😄 

So, first you noticed that the "onRequest/onError" part of the old Rule heavily depend on "instanceof" of ES-specific classes.

Therefore you split the "onResponse/onError" part from the "match" part and you put the first inside the es-dependent package.

The "match" part instead is not es-dependent because of "RequestContext" abstraction. So that could go together with generic code.

Now you are left with two aspects of the same thing: "RuleActionListener" implementors vs "Rule" implementors that cannot live in the same package. But both need to use the settings, so  you invented the RuleSettings interface.*

PS: *actually RuleActionListener cannot access its corresponding RuleSettings implementor. Shouldn't it?

So, I totally agree on where you came from with this design, it's a good solution for the current horrible state of the "onResponse/onError" logic, so es-specific.

But I can't help seeing here that **in the future** we could introduce a "ResponseContext" contract that would decouple the listener side like "RequestContext" did for the other part.

So yes, today this is good, but one day, once and if this is gonna be reasonable, we could re-consolidate the rule code and make it simple again.